### PR TITLE
Support PVR providers

### PR DESCRIPTION
**Pull request by phunkyfish**
v20.3.0
- Add icon add-on
- Add icons for providers
- Add Support for PVR Providers

This shows how to use providers in the demo add-on. Currently providers are only visible in the UI under recordings for demonstration purposes. 

The fallback in the UI can be seen here: https://github.com/xbmc/xbmc/pull/19395/commits/dbe2350b01dd353a8c949e715ed64416271457d2

Future work in Kodi is to support providers fully in the UI for channels and recordings. I'm sure @ksooo will have some ideas on how to best do this!

### CODE REVIEW

**Review by AlwinEsch — COMMENTED** [review]
From code it looks really good and not seen any problematic places :+1: 

Will look and test later, after I'm become back related Kodi for test.

File: src/PVRDemo.cpp [hunk 1]
```diff
@@ -186,6 +213,9 @@ PVR_ERROR CPVRDemo::GetChannels(bool bRadio, kodi::addon::PVRChannelsResultSet&
       kodiChannel.SetIsHidden(false);
       kodiChannel.SetHasArchive(channel.bArchive);
 
+      /* PVR API 8.0.0 */
+      kodiChannel.SetClientProviderUid(channel.iProviderId);
```
Comment (AlwinEsch) [inline src/PVRDemo.cpp:217]: Becomes given provider somewhere visible in channels list?

Seen only by recordings.

**Review by AlwinEsch — APPROVED** [review]
On test it has worked!

Only comes one question about the channels (see note there).

File: src/PVRDemo.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/PVRDemo.cpp:217]: Not yet. As the channels UI was recently reworked by @ksooo I thought it best if he could add the support there. My skinning foo is weak!

**Pull request merged**